--- a/configure	2017-02-08 14:12:50.626235483 -0500
+++ b/configure	2017-02-08 14:12:21.647626628 -0500
@@ -13608,6 +13608,7 @@
  esac
 
 
+if test "$cross_compiling" = no; then : 
 
 # Check whether --with-distro was given.
 if test "${with_distro+set}" = set; then :
@@ -13979,6 +13980,7 @@
 _ACEOF
 
 fi
+fi # end check if ! cross_compile
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking Linux Wireless Extensions >= 18" >&5
 $as_echo_n "checking Linux Wireless Extensions >= 18... " >&6; }
