--- a/libnm-util/Makefile
+++ b/libnm-util/Makefile
@@ -149,7 +149,9 @@
 am_libtest_crypto_la_OBJECTS = libtest_crypto_la-crypto.lo \
 	$(am__objects_4) $(am__objects_5)
 libtest_crypto_la_OBJECTS = $(am_libtest_crypto_la_OBJECTS)
-DEFAULT_INCLUDES = -I. -I$(top_builddir)
+DEFAULT_INCLUDES = -I. -I$(top_builddir) \
+-I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr \
+-I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 am__mv = mv -f
@@ -241,7 +243,7 @@
 CYGPATH_W = echo
 DATADIRNAME = share
 DBUS_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/dbus-1.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/dbus-1.0/include -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  
-DBUS_LIBS = -pthread -ldbus-glib-1 -ldbus-1 -lgobject-2.0 -lgthread-2.0 -lrt -lglib-2.0  
+DBUS_LIBS = -pthread -ldbus-glib-1 -ldbus-1 -lgobject-2.0 -lgthread-2.0 -lrt -lglib-2.0 -lnss3 -lnspr4 -lsmime3
 DBUS_SYS_DIR = /etc/dbus-1/system.d
 DEFS = -DHAVE_CONFIG_H
 DEPDIR = .deps
@@ -329,7 +331,7 @@
 PKG_CONFIG = /home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/bin/pkg-config
 POFILES = 
 POLKIT_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/polkit-1 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  
-POLKIT_LIBS = -pthread -lpolkit-gobject-1 -lgio-2.0 -lgobject-2.0 -lgmodule-2.0 -lgthread-2.0 -lrt -lglib-2.0  
+POLKIT_LIBS = -pthread -lpolkit-gobject-1 -lgio-2.0 -lgobject-2.0 -lgmodule-2.0 -lgthread-2.0 -lrt -lglib-2.0 -lnspr4 -lnss3 -lsmime3 
 POSUB = po
 PO_IN_DATADIR_FALSE = 
 PO_IN_DATADIR_TRUE = 
@@ -399,7 +401,7 @@
 top_build_prefix = ../
 top_builddir = ..
 top_srcdir = ..
-SUBDIRS = . tests
+SUBDIRS = . 
 INCLUDES = -I${top_srcdir} -I${top_srcdir}/include
 lib_LTLIBRARIES = libnm-util.la
 libnm_util_la_CPPFLAGS = $(GLIB_CFLAGS) $(DBUS_CFLAGS) $(UUID_CFLAGS) \
--- a/libnm-glib/Makefile
+++ b/libnm-glib/Makefile
@@ -37,7 +37,7 @@
 POST_UNINSTALL = :
 build_triplet = i686-pc-linux-gnu
 host_triplet = arm-unknown-linux-gnu
-noinst_PROGRAMS = libnm-glib-test$(EXEEXT)
+#noinst_PROGRAMS = libnm-glib-test$(EXEEXT)
 subdir = libnm-glib
 DIST_COMMON = $(libnminclude_HEADERS) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(srcdir)/libnm-glib-vpn.pc.in \
@@ -131,7 +131,7 @@
 	$(CFLAGS) $(libnm_glib_la_LDFLAGS) $(LDFLAGS) -o $@
 PROGRAMS = $(noinst_PROGRAMS)
 am_libnm_glib_test_OBJECTS =  \
-	libnm_glib_test-libnm-glib-test.$(OBJEXT)
+#	libnm_glib_test-libnm-glib-test.$(OBJEXT)
 libnm_glib_test_OBJECTS = $(am_libnm_glib_test_OBJECTS)
 libnm_glib_test_DEPENDENCIES = libnm-glib.la \
 	$(top_builddir)/libnm-util/libnm-util.la $(am__DEPENDENCIES_1) \
@@ -488,7 +488,7 @@
 libnm_glib_la_LDFLAGS = -Wl,--version-script=$(srcdir)/libnm-glib.ver \
 	-version-info "6:1:4"
 
-libnm_glib_test_SOURCES = libnm-glib-test.c
+#libnm_glib_test_SOURCES = libnm-glib-test.c
 libnm_glib_test_CFLAGS = $(GLIB_CFLAGS) $(DBUS_CFLAGS)
 libnm_glib_test_LDADD = libnm-glib.la $(top_builddir)/libnm-util/libnm-util.la $(GLIB_LIBS) $(DBUS_LIBS)
 libnm_glib_vpn_la_SOURCES = nm-vpn-plugin.c nm-vpn-plugin-ui-interface.c
@@ -596,9 +596,9 @@
 	list=`for p in $$list; do echo "$$p"; done | sed 's/$(EXEEXT)$$//'`; \
 	echo " rm -f" $$list; \
 	rm -f $$list
-libnm-glib-test$(EXEEXT): $(libnm_glib_test_OBJECTS) $(libnm_glib_test_DEPENDENCIES) 
-	@rm -f libnm-glib-test$(EXEEXT)
-	$(AM_V_CCLD)$(libnm_glib_test_LINK) $(libnm_glib_test_OBJECTS) $(libnm_glib_test_LDADD) $(LIBS)
+#libnm-glib-test$(EXEEXT): $(libnm_glib_test_OBJECTS) $(libnm_glib_test_DEPENDENCIES) 
+#	@rm -f libnm-glib-test$(EXEEXT)
+#	$(AM_V_CCLD)$(libnm_glib_test_LINK) $(libnm_glib_test_OBJECTS) $(libnm_glib_test_LDADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
@@ -634,7 +634,7 @@
 include ./$(DEPDIR)/libnm_glib_la-nm-settings-system-interface.Plo
 include ./$(DEPDIR)/libnm_glib_la-nm-types.Plo
 include ./$(DEPDIR)/libnm_glib_la-nm-vpn-connection.Plo
-include ./$(DEPDIR)/libnm_glib_test-libnm-glib-test.Po
+#include ./$(DEPDIR)/libnm_glib_test-libnm-glib-test.Po
 include ./$(DEPDIR)/libnm_glib_vpn_la-nm-vpn-plugin-ui-interface.Plo
 include ./$(DEPDIR)/libnm_glib_vpn_la-nm-vpn-plugin.Plo
 
@@ -905,17 +905,17 @@
 #	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) \
 #	$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libnm_glib_la_CFLAGS) $(CFLAGS) -c -o libnm_glib_la-nm-settings-service.lo `test -f 'nm-settings-service.c' || echo '$(srcdir)/'`nm-settings-service.c
 
-libnm_glib_test-libnm-glib-test.o: libnm-glib-test.c
-	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libnm_glib_test_CFLAGS) $(CFLAGS) -MT libnm_glib_test-libnm-glib-test.o -MD -MP -MF $(DEPDIR)/libnm_glib_test-libnm-glib-test.Tpo -c -o libnm_glib_test-libnm-glib-test.o `test -f 'libnm-glib-test.c' || echo '$(srcdir)/'`libnm-glib-test.c
-	$(AM_V_at)$(am__mv) $(DEPDIR)/libnm_glib_test-libnm-glib-test.Tpo $(DEPDIR)/libnm_glib_test-libnm-glib-test.Po
+#libnm_glib_test-libnm-glib-test.o: libnm-glib-test.c
+#	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libnm_glib_test_CFLAGS) $(CFLAGS) -MT libnm_glib_test-libnm-glib-test.o -MD -MP -MF $(DEPDIR)/libnm_glib_test-libnm-glib-test.Tpo -c -o libnm_glib_test-libnm-glib-test.o `test -f 'libnm-glib-test.c' || echo '$(srcdir)/'`libnm-glib-test.c
+#	$(AM_V_at)$(am__mv) $(DEPDIR)/libnm_glib_test-libnm-glib-test.Tpo $(DEPDIR)/libnm_glib_test-libnm-glib-test.Po
 #	$(AM_V_CC) \
 #	source='libnm-glib-test.c' object='libnm_glib_test-libnm-glib-test.o' libtool=no \
 #	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) \
 #	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libnm_glib_test_CFLAGS) $(CFLAGS) -c -o libnm_glib_test-libnm-glib-test.o `test -f 'libnm-glib-test.c' || echo '$(srcdir)/'`libnm-glib-test.c
 
-libnm_glib_test-libnm-glib-test.obj: libnm-glib-test.c
-	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libnm_glib_test_CFLAGS) $(CFLAGS) -MT libnm_glib_test-libnm-glib-test.obj -MD -MP -MF $(DEPDIR)/libnm_glib_test-libnm-glib-test.Tpo -c -o libnm_glib_test-libnm-glib-test.obj `if test -f 'libnm-glib-test.c'; then $(CYGPATH_W) 'libnm-glib-test.c'; else $(CYGPATH_W) '$(srcdir)/libnm-glib-test.c'; fi`
-	$(AM_V_at)$(am__mv) $(DEPDIR)/libnm_glib_test-libnm-glib-test.Tpo $(DEPDIR)/libnm_glib_test-libnm-glib-test.Po
+#libnm_glib_test-libnm-glib-test.obj: libnm-glib-test.c
+#	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libnm_glib_test_CFLAGS) $(CFLAGS) -MT libnm_glib_test-libnm-glib-test.obj -MD -MP -MF $(DEPDIR)/libnm_glib_test-libnm-glib-test.Tpo -c -o libnm_glib_test-libnm-glib-test.obj `if test -f 'libnm-glib-test.c'; then $(CYGPATH_W) 'libnm-glib-test.c'; else $(CYGPATH_W) '$(srcdir)/libnm-glib-test.c'; fi`
+#	$(AM_V_at)$(am__mv) $(DEPDIR)/libnm_glib_test-libnm-glib-test.Tpo $(DEPDIR)/libnm_glib_test-libnm-glib-test.Po
 #	$(AM_V_CC) \
 #	source='libnm-glib-test.c' object='libnm_glib_test-libnm-glib-test.obj' libtool=no \
 #	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) \
--- a/src/supplicant-manager/Makefile
+++ b/src/supplicant-manager/Makefile
@@ -312,7 +312,7 @@
 top_build_prefix = ../../
 top_builddir = ../..
 top_srcdir = ../..
-SUBDIRS = . tests
+SUBDIRS = .
 INCLUDES = \
 	-I${top_srcdir}/src \
 	-I${top_srcdir}/src/logging \
