--- a/src/system-settings/nm-sysconfig-settings.c	2010-08-09 18:50:55.000000000 -0400
+++ b/src/system-settings/nm-sysconfig-settings.c	2017-04-03 13:51:38.258167322 -0400
@@ -1505,10 +1505,11 @@
 nm_sysconfig_settings_init (NMSysconfigSettings *self)
 {
 	NMSysconfigSettingsPrivate *priv = NM_SYSCONFIG_SETTINGS_GET_PRIVATE (self);
+	GError *error;
 
 	priv->connections = g_hash_table_new_full (g_direct_hash, g_direct_equal, g_object_unref, NULL);
 
-	priv->authority = polkit_authority_get ();
+	priv->authority = polkit_authority_get_sync (NULL, &error);
 	if (priv->authority) {
 		priv->auth_changed_id = g_signal_connect (priv->authority,
 		                                          "changed",
--- a/libnm-util/Makefile	2017-04-04 14:11:43.925223953 -0400
+++ b/libnm-util/Makefile	2017-04-03 11:20:54.000000000 -0400
@@ -240,7 +240,7 @@
 CPPFLAGS = 
 CYGPATH_W = echo
 DATADIRNAME = share
-DBUS_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/dbus-1.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/dbus-1.0/include -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  
+DBUS_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/dbus-1.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/dbus-1.0/include -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr 
 DBUS_LIBS = -pthread -ldbus-glib-1 -ldbus-1 -lgobject-2.0 -lgthread-2.0 -lrt -lglib-2.0  
 DBUS_SYS_DIR = /etc/dbus-1/system.d
 DEFS = -DHAVE_CONFIG_H
@@ -257,22 +257,22 @@
 EXEEXT = 
 FGREP = /bin/grep -F
 GETTEXT_PACKAGE = NetworkManager
-GIO_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  
+GIO_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr 
 GIO_LIBS = -pthread -lgio-2.0 -lgobject-2.0 -lgmodule-2.0 -lgthread-2.0 -lrt -lglib-2.0  
-GLIB_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  
+GLIB_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr 
 GLIB_GENMARSHAL = glib-genmarshal
 GLIB_LIBS = -pthread -lgobject-2.0 -lgthread-2.0 -lrt -lglib-2.0  
-GMODULE_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  
+GMODULE_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr 
 GMODULE_LIBS = -Wl,--export-dynamic -pthread -lgmodule-2.0 -lrt -lglib-2.0  
 GMOFILES = 
 GMSGFMT = /usr/bin/msgfmt
-GNUTLS_CFLAGS = 
+GNUTLS_CFLAGS =  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr
 GNUTLS_LIBS = 
 GREP = /bin/grep
 GTKDOC_CHECK = 
 GTKDOC_MKPDF = 
 GTKDOC_REBASE = true
-GUDEV_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/gudev-1.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  
+GUDEV_CFLAGS = -pthread -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/gudev-1.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/glib-2.0 -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/lib/glib-2.0/include  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr 
 GUDEV_LIBS = -pthread -lgudev-1.0 -lgobject-2.0 -lgthread-2.0 -lrt -lglib-2.0  
 HTML_DIR = ${datadir}/gtk-doc/html
 INSTALL = /usr/bin/install -c
@@ -291,11 +291,11 @@
 LD = /home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/bin/arm-none-linux-gnueabi-ld
 LDFLAGS = 
 LIBDL = -ldl
-LIBGCRYPT_CFLAGS = 
+LIBGCRYPT_CFLAGS =  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr
 LIBGCRYPT_CONFIG = 
 LIBGCRYPT_LIBS = 
 LIBM = -lm
-LIBNL_CFLAGS =  
+LIBNL_CFLAGS =  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr 
 LIBNL_LIBS = -lnl  
 LIBOBJS = 
 LIBS = 
@@ -312,8 +312,9 @@
 MSGMERGE = /usr/bin/msgmerge
 NM = /home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/bin/arm-none-linux-gnueabi-nm
 NMEDIT = 
-NSS_CFLAGS = 
-NSS_LIBS = 
+NSS_CFLAGS =  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr 
+NSS_LIBS = -lnss3 -lnssutil3 -lsmime3 -lnss -lnssckbi -lnssdbm3 -lssl3 -lfreebl3 -lsoftokn3 
+NSPR_LIBS= -lnspr
 OBJDUMP = /home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/bin/arm-none-linux-gnueabi-objdump
 OBJEXT = o
 OTOOL = 
@@ -343,7 +344,7 @@
 SYSTEM_CA_PATH = /etc/ssl/certs
 UDEV_BASE_DIR = /lib/udev
 USE_NLS = yes
-UUID_CFLAGS = -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/uuid  
+UUID_CFLAGS = -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/uuid  -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nss -I/home/robert/50G/br_oct19/buildroot_2011_05/output/host/usr/arm-unknown-linux-gnueabi/sysroot/usr/include/nspr 
 UUID_LIBS = -luuid  
 VERSION = 0.8.1
 XGETTEXT = /usr/bin/xgettext
@@ -435,7 +436,7 @@
 	nm-setting-vpn.c nm-utils.c nm-utils-private.h \
 	$(libnm_util_include_HEADERS) $(am__append_1) $(am__append_4)
 libnm_util_la_LIBADD = $(GLIB_LIBS) $(DBUS_LIBS) $(UUID_LIBS) \
-	$(am__append_3) $(am__append_6)
+	$(am__append_3) $(am__append_6) 
 libnm_util_la_LDFLAGS = -Wl,--version-script=$(srcdir)/libnm-util.ver \
 	-version-info "7:0:6"
 
--- a/network-manager.c	2017-04-11 08:08:56.443763045 -0400
+++ b/network-manager.c	2017-04-11 08:47:09.503226220 -0400
@@ -23,7 +23,7 @@
 
 #include <glib.h>
 #include <glib/gi18n.h>
-#include <nm-client.h>
+#include "libnm-glib/nm-client.h"
 #include <nm-setting-connection.h>
 
 #include "utils.h"
--- a/cli/src/settings.c	2017-04-11 07:48:57.223886581 -0400
+++ b/cli/src/settings.c	2017-04-11 07:48:53.563870285 -0400
@@ -21,6 +21,7 @@
 #include <glib/gi18n.h>
 #include <libnm-util/nm-utils.h>
 
+#include "libnm-glib/nm-client.h"
 #include "utils.h"
 #include "settings.h"
 
--- a/cli/src/utils.c	2017-04-11 07:58:48.644067857 -0400
+++ b/cli/src/utils.c	2017-04-11 08:01:40.113223618 -0400
@@ -23,6 +23,7 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 
+#include "libnm-glib/nm-client.h"
 #include "utils.h"
 
 int
--- cli/src/nmcli.c	2017-04-11 08:03:33.281935097 -0400
+++ cli/src/nmcli.c	2017-04-11 08:07:04.267160919 -0400
@@ -32,6 +32,7 @@
 #include <glib/gi18n.h>
 #include <dbus/dbus.h>
 #include <dbus/dbus-glib.h>
+#include "libnm-glib/nm-client.h"
 #include <nm-client.h>
 #include <nm-setting-connection.h>
 #include <nm-remote-settings.h>
--- a/nm-client.h	2017-04-11 08:37:29.049965366 -0400
+++ b/nm-client.h	2017-04-11 08:37:38.598053529 -0400
@@ -28,9 +28,9 @@
 #include <glib-object.h>
 #include <dbus/dbus-glib.h>
 #include <NetworkManager.h>
-#include "nm-object.h"
-#include "nm-device.h"
-#include "nm-active-connection.h"
+#include "libnm-glib/nm-object.h"
+#include "libnm-glib/nm-device.h"
+#include "libnm-glib/nm-active-connection.h"
 
 G_BEGIN_DECLS
 
--- a/Makefile	2017-04-11 09:08:58.088893072 -0400
+++ b/Makefile	2017-04-11 09:09:12.236967474 -0400
@@ -328,7 +328,6 @@
 	tools \
 	policy \
 	initscript \
-	test \
 	po \
 	man \
 	docs \
@@ -341,7 +340,7 @@
 	intltool-merge.in \
 	intltool-update.in		
 
-DISTCHECK_CONFIGURE_FLAGS = --with-tests=yes --with-docs=yes --with-udev-dir=$$dc_install_base/lib/udev
+DISTCHECK_CONFIGURE_FLAGS = --with-tests=no --with-docs=no --with-udev-dir=$$dc_install_base/lib/udev
 DISTCLEANFILES = intltool-extract intltool-merge intltool-update
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = NetworkManager.pc
--- a/test/nm-tool.c	2010-08-06 14:59:06.000000000 -0400
+++ b/test/nm-tool.c	2017-04-04 09:24:15.013419376 -0400
@@ -31,17 +31,18 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 
-#include <nm-client.h>
-#include <nm-device.h>
-#include <nm-device-ethernet.h>
-#include <nm-device-wifi.h>
-#include <nm-gsm-device.h>
-#include <nm-cdma-device.h>
-#include <nm-device-bt.h>
+#include <libnm-glib/nm-access-point.h>
+#include <libnm-glib/nm-client.h>
+#include <libnm-glib/nm-device.h>
+#include <libnm-glib/nm-device-ethernet.h>
+#include <libnm-glib/nm-device-wifi.h>
+#include <libnm-glib/nm-gsm-device.h>
+#include <libnm-glib/nm-cdma-device.h>
+#include <libnm-glib/nm-device-bt.h>
 #include <nm-utils.h>
 #include <nm-setting-ip4-config.h>
 #include <nm-setting-ip6-config.h>
-#include <nm-vpn-connection.h>
+#include <libnm-glib/nm-vpn-connection.h>
 #include <nm-setting-connection.h>
 
 /* Don't use nm-dbus-glib-types.h so that we can keep nm-tool
--- a/cli/src/connections.c	2010-08-06 14:59:04.000000000 -0400
+++ b/cli/src/connections.c	2017-04-04 09:17:04.290109137 -0400
@@ -28,8 +28,10 @@
 #include <errno.h>
 #include <signal.h>
 #include <netinet/ether.h>
+#include <libnm-glib/nm-device-ethernet.h>
+#include <libnm-glib/nm-device-wifi.h>
 
-#include <nm-client.h>
+#include <libnm-glib/nm-client.h>
 #include <nm-setting-connection.h>
 #include <nm-setting-wired.h>
 #include <nm-setting-pppoe.h>
@@ -38,17 +40,14 @@
 #include <nm-setting-cdma.h>
 #include <nm-setting-bluetooth.h>
 #include <nm-setting-olpc-mesh.h>
-#include <nm-device-ethernet.h>
-#include <nm-device-wifi.h>
-#include <nm-gsm-device.h>
-#include <nm-cdma-device.h>
-#include <nm-device-bt.h>
-//#include <nm-device-olpc-mesh.h>
-#include <nm-remote-settings.h>
-#include <nm-remote-settings-system.h>
-#include <nm-settings-interface.h>
-#include <nm-settings-connection-interface.h>
-#include <nm-vpn-connection.h>
+#include <libnm-glib/nm-gsm-device.h>
+#include <libnm-glib/nm-cdma-device.h>
+#include <libnm-glib/nm-device-bt.h>
+#include <libnm-glib/nm-remote-settings.h>
+#include <libnm-glib/nm-remote-settings-system.h>
+#include <libnm-glib/nm-settings-interface.h>
+#include <libnm-glib/nm-settings-connection-interface.h>
+#include <libnm-glib/nm-vpn-connection.h>
 
 #include "utils.h"
 #include "settings.h"
--- a/cli/src/devices.c	2010-08-06 14:59:04.000000000 -0400
+++ b/cli/src/devices.c	2017-04-04 08:49:01.414221905 -0400
@@ -26,23 +26,23 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 
+#include "libnm-glib/nm-ip4-config.h"
+#include "libnm-glib/nm-ip6-config.h"
+#include <libnm-glib/nm-device.h>
 #include <glib.h>
 #include <glib/gi18n.h>
-#include <nm-client.h>
-#include <nm-device-wifi.h>
+#include <libnm-glib/nm-device-ethernet.h>
+#include <libnm-glib/nm-device-wifi.h>
+#include <libnm-glib/nm-client.h>
+#include <libnm-glib/nm-access-point.h>
 
-#include <nm-client.h>
-#include <nm-device.h>
-#include <nm-device-ethernet.h>
-#include <nm-device-wifi.h>
-#include <nm-gsm-device.h>
-#include <nm-cdma-device.h>
-#include <nm-device-bt.h>
-//#include <nm-device-olpc-mesh.h>
+#include <libnm-glib/nm-gsm-device.h>
+#include <libnm-glib/nm-cdma-device.h>
+#include <libnm-glib/nm-device-bt.h>
 #include <nm-utils.h>
 #include <nm-setting-ip4-config.h>
 #include <nm-setting-ip6-config.h>
-#include <nm-vpn-connection.h>
+#include <libnm-glib/nm-vpn-connection.h>
 #include <nm-setting-connection.h>
 #include <nm-setting-wired.h>
 #include <nm-setting-pppoe.h>
@@ -55,7 +55,6 @@
 #include "utils.h"
 #include "devices.h"
 
-
 /* Available fields for 'dev status' */
 static NmcOutputField nmc_fields_dev_status[] = {
 	{"DEVICE",  N_("DEVICE"),    10, NULL, 0},  /* 0 */
--- a/src/Makefile	2017-04-03 11:20:46.000000000 -0400
+++ b/src/Makefile	2017-04-03 14:51:18.217161006 -0400
@@ -423,7 +423,6 @@
 	bluez-manager \
 	system-settings \
 	. \
-	tests
 
 INCLUDES = -I${top_srcdir}                   \
            -I${top_srcdir}/include           \
--- a/src/nm-device-ethernet.c	2010-08-10 01:31:46.000000000 -0400
+++ b/src/nm-device-ethernet.c	2017-04-03 14:50:39.908917630 -0400
@@ -307,7 +307,7 @@
 	const char *iface;
 	GUdevClient *client;
 	GUdevDevice *dev;
-	GUdevDevice *parent;
+	GUdevDevice *parent=NULL;
 	const char *parent_path, *item, *driver;
 	const char *subsystems[] = { "net", NULL };
 	GDir *dir;
--- a/src/nm-manager.c	2010-08-10 01:31:39.000000000 -0400
+++ b/src/nm-manager.c	2017-04-03 14:44:18.337179235 -0400
@@ -4197,6 +4197,7 @@
 	DBusGConnection *g_connection;
 	guint id, i;
 	GFile *file;
+	GError *error; 
 
 	/* Initialize rfkill structures and states */
 	memset (priv->radio_states, 0, sizeof (priv->radio_states));
@@ -4279,7 +4280,7 @@
 	} else
 		nm_log_warn (LOGD_AUTOIP4, "could not initialize avahi-autoipd D-Bus proxy");
 
-	priv->authority = polkit_authority_get ();
+	priv->authority = polkit_authority_get_sync (NULL, &error);
 	if (priv->authority) {
 		priv->auth_changed_id = g_signal_connect (priv->authority,
 		                                          "changed",
--- a/nm-vpn-connection.h	2010-08-06 14:59:05.000000000 -0400
+++ b/nm-vpn-connection.h	2017-04-03 14:30:59.710902601 -0400
@@ -1,24 +1,22 @@
 /* -*- Mode: C; tab-width: 4; indent-tabs-mode: t; c-basic-offset: 4 -*- */
-/*
- * libnm_glib -- Access network status & information from glib applications
+/* NetworkManager -- Network link manager
  *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public
- * License as published by the Free Software Foundation; either
- * version 2 of the License, or (at your option) any later version.
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
  *
- * This library is distributed in the hope that it will be useful,
+ * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
  *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library; if not, write to the
- * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
- * Boston, MA 02110-1301 USA.
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  *
- * Copyright (C) 2007 - 2008 Novell, Inc.
- * Copyright (C) 2007 - 2010 Red Hat, Inc.
+ * Copyright (C) 2005 - 2008 Red Hat, Inc.
+ * Copyright (C) 2006 - 2008 Novell, Inc.
  */
 
 #ifndef NM_VPN_CONNECTION_H
@@ -26,11 +24,10 @@
 
 #include <glib.h>
 #include <glib-object.h>
-#include <dbus/dbus-glib.h>
-#include "nm-active-connection.h"
 #include "NetworkManagerVPN.h"
-
-G_BEGIN_DECLS
+#include "nm-device.h"
+#include "nm-activation-request.h"
+#include "nm-secrets-provider-interface.h"
 
 #define NM_TYPE_VPN_CONNECTION            (nm_vpn_connection_get_type ())
 #define NM_VPN_CONNECTION(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), NM_TYPE_VPN_CONNECTION, NMVPNConnection))
@@ -43,33 +40,39 @@
 #define NM_VPN_CONNECTION_BANNER "banner"
 
 typedef struct {
-	NMActiveConnection parent;
+	GObject parent;
 } NMVPNConnection;
 
 typedef struct {
-	NMActiveConnectionClass parent;
+	GObjectClass parent;
 
 	/* Signals */
 	void (*vpn_state_changed) (NMVPNConnection *connection,
 	                           NMVPNConnectionState state,
 	                           NMVPNConnectionStateReason reason);
 
-	/* Padding for future expansion */
-	void (*_reserved1) (void);
-	void (*_reserved2) (void);
-	void (*_reserved3) (void);
-	void (*_reserved4) (void);
-	void (*_reserved5) (void);
-	void (*_reserved6) (void);
+	void (*properties_changed) (NMVPNConnection *connection, GHashTable *properties);
 } NMVPNConnectionClass;
 
 GType nm_vpn_connection_get_type (void);
 
-GObject * nm_vpn_connection_new (DBusGConnection *dbus_connection, const char *path);
-
-NMVPNConnectionState  nm_vpn_connection_get_vpn_state  (NMVPNConnection *vpn);
-const char *          nm_vpn_connection_get_banner (NMVPNConnection *vpn);
-
-G_END_DECLS
+NMVPNConnection * nm_vpn_connection_new (NMConnection *connection,
+                                         NMActRequest *act_request,
+                                         NMDevice *parent_device);
+
+void                 nm_vpn_connection_activate        (NMVPNConnection *connection);
+NMConnection *       nm_vpn_connection_get_connection  (NMVPNConnection *connection);
+const char *         nm_vpn_connection_get_active_connection_path (NMVPNConnection *connection);
+const char *         nm_vpn_connection_get_name        (NMVPNConnection *connection);
+NMVPNConnectionState nm_vpn_connection_get_vpn_state   (NMVPNConnection *connection);
+const char *         nm_vpn_connection_get_banner      (NMVPNConnection *connection);
+void                 nm_vpn_connection_fail            (NMVPNConnection *connection,
+                                                        NMVPNConnectionStateReason reason);
+void                 nm_vpn_connection_disconnect      (NMVPNConnection *connection,
+                                                        NMVPNConnectionStateReason reason);
+NMIP4Config *        nm_vpn_connection_get_ip4_config  (NMVPNConnection *connection);
+const char *         nm_vpn_connection_get_ip_iface    (NMVPNConnection *connection);
+NMDevice *           nm_vpn_connection_get_parent_device (NMVPNConnection *connection);
+guint32              nm_vpn_connection_get_ip4_internal_gateway (NMVPNConnection *connection);
 
 #endif /* NM_VPN_CONNECTION_H */
--- a/src/system-settings/nm-sysconfig-connection.c	2010-08-06 14:59:06.000000000 -0400
+++ b/src/system-settings/nm-sysconfig-connection.c	2017-04-03 13:50:45.585822635 -0400
@@ -612,8 +612,9 @@
 nm_sysconfig_connection_init (NMSysconfigConnection *self)
 {
 	NMSysconfigConnectionPrivate *priv = NM_SYSCONFIG_CONNECTION_GET_PRIVATE (self);
+	GError *error;
 
-	priv->authority = polkit_authority_get ();
+	priv->authority = polkit_authority_get_sync (NULL, &error);
 	if (!priv->authority) {
 		nm_log_err (LOGD_SYS_SET, "%s: error creating PolicyKit authority");
 	}
