--- a/autogen.sh	2016-12-22 20:38:26.000000000 -0500
+++ b/autogen.sh	2017-01-17 15:56:39.323454654 -0500
@@ -24,14 +24,7 @@
         echo "Activated pre-commit hook." || :
 fi
 
-if which gtkdocize >/dev/null 2>/dev/null; then
-        gtkdocize --docdir docs/
-        gtkdocargs=--enable-gtk-doc
-else
-        echo "You don't have gtk-doc installed, and thus won't be able to generate the documentation."
-        rm -f docs/gtk-doc.make
-        echo 'EXTRA_DIST =' > docs/gtk-doc.make
-fi
+# gtkdocargs=--disable-gtk-doc
 
 intltoolize --force --automake
 autoreconf --force --install --symlink
