--- a/docs/libudev/Makefile.am	2016-12-22 20:38:26.000000000 -0500
+++ b/docs/libudev/Makefile.am	2017-01-18 16:03:01.196307421 -0500
@@ -89,11 +89,11 @@
 GTKDOC_LIBS=
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/docs/gtk-doc.make
+# include $(top_srcdir)/docs/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += version.xml.in
+EXTRA_DIST = version.xml.in
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
