--- a/arch/powerpc/platforms/52xx/Kconfig	2010-05-16 17:17:36.000000000 -0400
+++ b/arch/powerpc/platforms/52xx/Kconfig	2018-01-16 15:41:15.837876984 -0500
@@ -36,6 +36,11 @@
 	select RTAS_PROC
 	select PPC_NATIVE
 
+config PPC_SPACECOM2
+	bool "B.Braun SpaceCom 2 Board"
+	help
+	Support for the B.Braun SpaceCom 2 Board.
+
 config PPC_LITE5200
 	bool "Freescale Lite5200 Eval Board"
 	depends on PPC_MPC52xx
--- a/kernel/timeconst.pl
+++ b/kernel/timeconst.pl
@@ -370,7 +370,7 @@
 	}
 
 	@val = @{$canned_values{$hz}};
-	if (!defined(@val)) {
+	if (!(@val)) {
 		@val = compute_values($hz);
 	}
 	output($hz, @val);
